src/utils/page.js

```diff
@@ -38,6 +41,27 @@ export function getPage (win = window) {
   return detectedPageUrl
 }
 
+/**
+ * @return {string|undefined}
+ */
+export function getContextElements (contextSelectors, contextElementsLength, win = window) {
+  if (!contextSelectors || contextSelectors === '') {
+    return ''
+  } else {
+    var collectedElements = _collectElementsText(contextSelectors, win)
+    return base64UrlEncode(collectedElements).slice(0, contextElementsLength)
```

Comment:
We also might think about only collecting as much data as we would eventually send by checking `4 * ceil(n/3.0) < contextElementsLength` where `n` is the number of bits in the utf-8 encoded accumulator for the elements added to it so far. It is debatable, but might prevent building huge strings that we would cut off either way.